packages/mermaid/src/diagrams/gantt/parser/gantt.spec.js

```diff
@@ -26,6 +26,11 @@ describe('when parsing a gantt diagram it', function () {
 
     expect(parserFnConstructor(str)).not.toThrow();
   });
+  it('should handle a dateRange definition', function () {
+    const str = 'gantt\ndateRange : 2023-06-01, 2023-07-01';
+
+    expect(parserFnConstructor(str)).not.toThrow();
+  });
```

Comment:
We need more test cases here. 

And the behavior below is flaky. The format mentioned in the docs is not tested here. 
So we should make it strict, then test and document it well. 

`dateRange 13:00, 14:00` fails
`dateRange  13:00, 14:00` passes (two spaces)
`dateRange: 13:00, 14:00` fails
`dateRange : 13:00, 14:00` passes

Reply:
I've made it more strict by removing the logic that would attempt to parse the date using `new Date` after `dayjs.isValid()` returns `false` (see [this commit](https://github.com/mermaid-js/mermaid/pull/5220/commits/81b57d6b2276f1200d5638e181165e2bd7648a62)). This change has not broken any tests, but I suspect it might break existing charts that are passing an invalid `dateFormat` while still successfully rendering due to that "backup" logic.

For valid date ranges, I've now included a test with a date-only range, a time only range, and a date+time range. Are there any other test cases I should add?

For the tests for invalid date ranges, I realised that depending on the specific error in syntax, the `dateRange` declaration could either be ignored or throw an error.

From your examples above, `dateRange: 13:00, 14:00` would be ignored since the `dateRange:` (without a space before the colon) would not be considered a match by the jison parser looking for `"dateRange"\s[^#\n;]`, and the `dateRange` would remain as the default of `''`. For `dateRange : 13:00, 14:00` (space before colon), the declaration would be picked up, but an error will be thrown as `: 13:00` would be considered an invalid date. I've added tests for both these scenarios (see [this commit](https://github.com/mermaid-js/mermaid/pull/5220/commits/0444adb25c7100cb9ae3667b440cd8caa7c8258e)), and would appreciate any feedback!